constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/SpansAlias.java

```diff
@@ -0,0 +1,31 @@
+package gov.nasa.jpl.aerie.constraints.tree;
+
+import gov.nasa.jpl.aerie.constraints.model.EvaluationEnvironment;
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+import gov.nasa.jpl.aerie.constraints.time.Segment;
+import gov.nasa.jpl.aerie.constraints.time.Spans;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+
+import java.util.Objects;
+import java.util.Set;
+
+public record SpansAlias(String spansAlias) implements Expression<Spans> {
```

Comment:
Based on the test added, am I correct in assuming that SpansAlias is a class for referring to a span by a name?

Reply:
Yes. The constraints package is essentially a very (very) basic unoptimized interpreter for the constraints AST. Unlike a basic Javascript or Python interpreter, we don't have a langauge (i.e. DSL) that we need to lex and parse into the AST; the constraints typescript API generates the AST directly. The `EvaluationEnvironment` record is how we store variables. The environment comes preloaded with resource bindings from simulation results - sort of like auto-imported global variables like `document` in browser-based js. Then during evaluation we can define more variables like activity aliases in `ForEachActivity`.

In this case, if a spans expression is being evaluated in the context of a Spans based scheduling goal, the `getConflicts` method will preload the environment with both the simulation results and a spans alias, which can be queried using the `SpansAlias` expression AST node.